src/framework/ui/internal/platform/linux/linuxplatformtheme.cpp

```diff
@@ -28,26 +28,45 @@
 using namespace mu::ui;
 using namespace mu::async;
 
+static const QLatin1String service("org.freedesktop.portal.Desktop");
+static const QLatin1String path("/org/freedesktop/portal/desktop");
+static const QLatin1String interface("org.freedesktop.portal.Settings");
+
 LinuxPlatformTheme::LinuxPlatformTheme()
 {
 }
 
 void LinuxPlatformTheme::startListening()
 {
+    QDBusConnection::sessionBus().connect(service, path, interface, QStringLiteral("SettingChanged"), this,
+                                          SLOT(processSettingChange(QString,QString,QDBusVariant)));
+    m_isSystemThemeDark = isSystemThemeCurrentlyDark();
+    m_isListening = true;
 }
 
 void LinuxPlatformTheme::stopListening()
 {
+    QDBusConnection::sessionBus().disconnect(service, path, interface, QStringLiteral("SettingChanged"), this,
+                                             SLOT(processSettingChange(QString,QString,QDBusVariant)));
+    m_isListening = false;
 }
 
 bool LinuxPlatformTheme::isFollowSystemThemeAvailable() const
 {
+    const QDBusConnection connection = QDBusConnection::sessionBus();
+    static const QString registrarService = QStringLiteral("org.freedesktop.portal.Desktop");
+    if (const auto iface = connection.interface()) {
+        return iface->isServiceRegistered(registrarService);
+    }
     return false;
```

Comment:
It is our convention to add some extra newlines for readability. For example, here...
```suggestion
    const QDBusConnection connection = QDBusConnection::sessionBus();
    
    if (const auto iface = connection.interface()) {
        static const QString registrarService = QStringLiteral("org.freedesktop.portal.Desktop");
        return iface->isServiceRegistered(registrarService);
    }
    
    return false;
```